### ISSUE TITLE
Default letter-key controls only react to upper-case letters and are sticky, lower-case letters are ignored by minetest

### ISSUE THREAD

**Issue by toomanychar** [minetest/minetest#13702]
##### Minetest version

```
Minetest 5.7.0 (Linux)
Using Irrlicht 1.9.0mt10
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="/usr/share/minetest"
STATIC_LOCALEDIR="/usr/share/locale"
```

Active renderer: OpenGL 4.6

##### OS / Hardware

Operating system: Linux Mint 21.1 Cinnamon 64-bit
CPU: AMD Ryzen 5 5600H with Radeon Graphics
I have the following keyboard layouts installed:
Russian
English (US)
German
I have attempted to launch minetest with each of these keyboard layouts selected, and I attempted to use all three of the layouts while in-game. The result does not change.

##### Summary

If you join any server/world you are unable to move, open your inventory, etc, unless you hold down Shift or activate Caps Lock. Basically, you can only move if you press upper-case control keys. Trying to change any control to a lower-case letter in the key-changing menu results in empty spaces (see screenshot). If you run minetest from a terminal, the game often outputs error messages such as the following:
``` EKEY_CODE not found, X11 keycode: 1731 ```
Strangely, it's possible to type lower-case letters in chat. Typing them in chat also results in many error messages similar to the one above.
Also, if you press upper-case letters, the game starts thinking you are HOLDING THEM DOWN. Basically, if you enable Caps Lock and press W, then the game will start thinking you are holding down W and make you move forward continuously. It will continue thinking that until you press Esc.

##### Steps to reproduce

1. Add Minetest Ubuntu Stable PPA
```sudo add-apt-repository ppa:minetestdevs/stable && sudo apt update```
2. Install minetest
```sudo apt install minetest```
3. Run minetest in terminal
```minetest```
4. Create new world, join it
5a. Press W by itself (without holding down any of the keys). This will result in you NOT moving.
5b. Hold Shift and press W, then stop holding Shift. This will result in you continuously moving forward. Pressing Esc or switching to another tab stops this.
5c. Go to the key-changing menu. Select any control and press W (or any other letter key) by itself. This will result in an empty field appearing.
![chaterror](https://github.com/minetest/minetest/assets/81852085/5f03eee1-8af1-45a1-812b-a41e44c16fd3)
![emptykeys](https://github.com/minetest/minetest/assets/81852085/55d0fa66-0876-4c71-a737-d9a91f5e1499)

**Comment by fluxionary** [comment]
> I have the following keyboard layouts installed:
> Russian
> English (US)
> German

this isn't an issue that affects most people. my guess is that it has something to do w/ whatever multi-script input mechanism you have installed. can you provide any details on what that is, including versions and sources?

**Comment by toomanychar** [comment]
I have no clue what you mean by "multi-script input mechanism". Do you mean the X-server?
What details specifically do you want, and how do I provide them?

**Comment by Zughy** [comment]
@toomanychar fluxionary means what do you use to change between languages on your OS. Is it something buliltin with Mint?

**Comment by grorp** [comment]
@toomanychar Also, does compiling Minetest with SDL2 (`cmake -DUSE_SDL2=TRUE`) change anything?

**Comment by toomanychar** [comment]
> @toomanychar fluxionary means what do you use to change between languages on your OS. Is it something buliltin with Mint?

I didn't change the input mechanism aside from adding layouts. Supposedly the applet is builtin with cinnamon, it's called "Keyboard (keyboard@cinnamon.org)".

**Comment by toomanychar** [comment]
> @toomanychar Also, does compiling Minetest with SDL2 (`cmake -DUSE_SDL2=TRUE`) change anything?

Trying that command results in an error.

I used the official guide on compiling on Linux - downloaded the dependencies using the given command, downloaded source, minetest game and Irrlicht without using Git using the given commands, and tried running the exact command you gave me. That resulted in this error:

Log file: [CMakeOutput.log](https://github.com/minetest/minetest/files/12806539/CMakeOutput.log)

```
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- *** Will build version 5.8.0-dev ***
-- Using user-provided IrrlichtMt at subdirectory 'lib/irrlichtmt'
-- *** Building IrrlichtMt 1.9.0.12 ***
-- Performing Test REVISION_SANITY_CHECK
-- Performing Test REVISION_SANITY_CHECK - Success
-- Device: SDL
-- OpenGL: ON
-- OpenGL 3: ON
-- OpenGL ES: OFF
-- OpenGL ES 2: OFF
-- EGL: OFF
-- WebGL: OFF
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11") 
-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so (found version "80") 
-- Found PNG: /usr/lib/x86_64-linux-gnu/libpng.so (found version "1.6.37") 
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
CMake Error at lib/irrlichtmt/source/Irrlicht/CMakeLists.txt:283 (find_package):
  Could not find a package configuration file provided by "SDL2" with any of
  the following names:

    SDL2Config.cmake
    sdl2-config.cmake

  Add the installation prefix of "SDL2" to CMAKE_PREFIX_PATH or set
  "SDL2_DIR" to a directory containing one of the above files.  If "SDL2"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
See also "/home/toomanycharacter/temp/minetest-master/CMakeFiles/CMakeOutput.log".
```

**Comment by Desour** [comment]
You'll have to install sdl2 to build with sdl2. It should be something like `sdl2-dev` or `libsdl2-dev`.

Just for linking, here's another input bindings issue: #13031

**Comment by toomanychar** [comment]
> You'll have to install sdl2 to build with sdl2. It should be something like `sdl2-dev` or `libsdl2-dev`.
> 
> Just for linking, here's another keyboard bindings issue: #13031

I already had libsdl2-2.0-0 installed, but alright, I installed libsdl2-dev.

As a test, I tried building minetest using the way in the guide. It compiled successfully and the exact same issues occured.

When compiling using `cmake -DUSE_SDL2=TRUE` and then `make -j$(nproc)`, the issue fixed itself. I was able to move and change keybindings without any problems, and no error messages appeared in the terminal.

I suppose the issue is solved by compiling with SDL2 then.
